frankenstein/ingester_lifecycle.go

```diff
@@ -30,23 +32,59 @@ const (
 	infName = "eth0"
 )
 
-// WriteIngesterConfigToConsul writes ingester config to Consul
-func WriteIngesterConfigToConsul(consulClient ConsulClient, listenPort int, numTokens int) error {
-	log.Info("Adding ingester to consul")
+// IngesterRegistration is the registration of an ingester with a register.
+type IngesterRegistration interface {
+	Unregister() error
+}
```

Comment:
Can you please give me a source for that adage?

In other languages, I'm a really big fan of using abstract types (which Golang interfaces seem to be) in public interfaces. I wish I had a write-up shorter than [this Youtube video](https://www.youtube.com/watch?v=3MNVP9-hglc).